doc/service_endpoint.md

```diff
@@ -0,0 +1,85 @@
+# Service Endpoint
+
+## Creating GitHub Service Endpoint
+
+Use command:
+
+``` bash
+az devops service-endpoint github create
+```
+
+In interactive mode this command will ask for [GitHub PAT token](https://help.github.com/en/articles/creating-a-personal-access-token-for-the-command-line) using a prompt message, for automation purpose set the GitHub PAT token in `AZURE_DEVOPS_EXT_GITHUB_PAT` environment variable.
+
+## Creating Azure RM Service Endpoint
+
+Use command:
+
+```bash
+az devops service-endpoint azurerm create
+```
+
+### Using client secret/password
+
+In interactive mode this command will ask for service principal password/secret using a prompt message, for automation purpose set service principal password/secret in `AZURE_DEVOPS_EXT_AZURE_RM_SERVICE_PRINCIPAL_KEY` environment variable.
+
+### Using client certificate
+
+If AAD application uses [certificate for authentification](https://docs.microsoft.com/en-us/azure/active-directory/develop/active-directory-certificate-credentials) then create .pem for the certificate and pass path to .pem file in `--azure-rm-service-principal-certificate-path` argument.
+
+.pem file can be created using openssl 
+
+```bash
+openssl pkcs12 -in file.pfx -out file.pem -nodes -password pass:<password_here>
+```
+
+## Create service endpoint using configuration point
```

Comment:
Nit: Can you start with this and then get to create github and azure rm service connection? General -> specific?

Reply:
as discussed offline, not changing